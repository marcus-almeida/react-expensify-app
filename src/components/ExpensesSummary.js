import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expensesTotal';
import getVisibleExpenses from '../selectors/expenses';


export class ExpensesSummary extends React.Component {

	render() {
		return (
            <div>
            <p>
                Viewing {this.props.expenseCount} {this.props.expenseCount === 1 ? 'expense' : 'expenses'} totaling {numeral(this.props.expensesTotal).format('$0,0.00')}
            </p>
			</div>
		);
	}
}


const mapStateToProps = (state) => ({ 
    expenseCount: getVisibleExpenses(state.expenses, state.filters).length,
    expensesTotal: getExpensesTotal(getVisibleExpenses(state.expenses, state.filters))
});


export default connect(mapStateToProps)(ExpensesSummary);
