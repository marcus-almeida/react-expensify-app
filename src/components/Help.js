import React from 'react';

const Help = () => (
	<div>
		this is from my help component
	</div>
);

export default Help;
