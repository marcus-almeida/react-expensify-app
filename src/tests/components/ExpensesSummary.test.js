import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={94.34} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={94.34} />);
    expect(wrapper).toMatchSnapshot();
});