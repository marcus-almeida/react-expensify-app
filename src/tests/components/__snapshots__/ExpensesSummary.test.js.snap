// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totaling 
    $94.34
  </p>
</div>
`;

exports[`should render ExpensesSummary 2`] = `
<div>
  <p>
    Viewing 
    2
     
    expenses
     totaling 
    $94.34
  </p>
</div>
`;
